.github/actions/install-java-environment/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Install Java Environment
+description: "Installs the Java environment"
+inputs:
+  java_version:
+    description: "Java version"
+    required: false
+    default: "21"
+    type: string
+  gradle_cache_read_only:
+    description: "Whether to use a read-only Gradle cache"
+    required: false
+    default: false
+    type: boolean
+  gradle_cache_write_only:
+    description: "Whether to use a write-only Gradle cache"
+    required: false
+    default: false
+    type: boolean
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/setup-java@v4
+      with:
+        distribution: corretto
+        java-version: ${{ inputs.java_version }}
+    - uses: gradle/actions/setup-gradle@v3
+      with:
+        cache-read-only: ${{ inputs.gradle_cache_read_only }}
+        cache-write-only: ${{ inputs.gradle_cache_write_only }}
+    - name: Install system tools (jq, xargs)
+      shell: bash
+      run: sudo apt-get -y install jq findutils
```

Comment:
Is this step really required? Have you tried without? jq is no longer needed (IIRC) and xargs should be present on the system (which is now a full fledged ubuntu and not some bare bones alpine distro)

Reply:
I removed it and it still works. Thanks for the suggestion